src/Shared/FileMatcher.cs

```diff
@@ -1905,21 +2123,42 @@ DirectoryExists directoryExists
              * This is because it's cheaper to add items to an IList and this code
              * might potentially do a lot of that.
              */
-            var listOfFiles = new List<string>();
+            var listOfFiles = new ConcurrentStack<string>();
 
             /*
              * Now get the files that match, starting at the lowest fixed directory.
              */
             try
             {
+                // Setup the values for calculating the MaxDegreeOfParallelism option of Parallel.ForEach
+                // Set to use only half processors when we have 4 or more of them, in order to not be too aggresive
+                // By setting MaxTasksPerIteration to the maximum amount of tasks, which means that only one
+                // Parallel.ForEach will run at once, we get a stable number of threads being created.
+                var maxTasks = Math.Max(1, Environment.ProcessorCount / 2);
```

Comment:
@lifengl What do you think about this when it runs inside VS?

Reply:
That seems to be fine to me.  In dev 14 time, during the solution loading time, we use about 2.2 cores inside VS, and design time build uses about 1 core in the background.  However, in .net core project, after we disabled the async tree loading, and with very long time in the evaluation, it falls back to almost one core, especially during the project evaluation phase.  So this change will make the evaluation more efficient during that phase.

In the project editing phase, or project reloading time, we may have multiple evaluation in the background.  Ideally, if we can control that based on the priority of the evaluation (foreground or background), it will be better.  But it looks like this setting is not too aggressive.